component/src/main/java/io/siddhi/extension/io/azurestorage/sink/DataLakeSink.java

```diff
@@ -0,0 +1,416 @@
+                                "For each event received to the sink, it will get appended to the file in csv format" +
+                                "output will looks like below.\n" +
+                                "WSO2,55.6,100\n" +
+                                "IBM,55.7,102"
+                )
+        }
+)
+public class DataLakeSink extends Sink {
+    private static final Log log = LogFactory.getLog(DataLakeSink.class);
+    private DataLakeFileSystemClient dataLakeFileSystemClient;
+    private String accountName;
+    private Option filePathOption;
+    private String blobContainer;
+    boolean appendIfFileExists;
+    private DataLakeFileClient fileClient;
+    private DataLakeServiceClient storageClient;
+    private boolean addLineSeparator;
+    private long fileOffset;
+    private SiddhiAppContext siddhiAppContext;
+    private boolean containerExist;
+    private Map<String, FileClientInfo> fileClientMap;
+    private FileClientInfo currentFileClientInfo;
+    private String currentFilePath;
+
+    @Override
+    public Class[] getSupportedInputEventClasses() {
+        return new Class[]{String.class, byte[].class, Object.class};
+    }
+
+    @Override
+    protected ServiceDeploymentInfo exposeServiceDeploymentInfo() {
+        return null;
+    }
+
+    @Override
+    public String[] getSupportedDynamicOptions() {
+        return new String[]{Constant.FILE_PATH};
+    }
+
+    @Override
+    protected StateFactory init(StreamDefinition streamDefinition, OptionHolder optionHolder, ConfigReader configReader,
+                                SiddhiAppContext siddhiAppContext) {
+        this.siddhiAppContext = siddhiAppContext;
+        accountName = optionHolder.validateAndGetStaticValue(Constant.ACCOUNT_NAME);
+        String accountKey = optionHolder.validateAndGetStaticValue(Constant.ACCOUNT_KEY);
+        blobContainer = optionHolder.validateAndGetStaticValue(Constant.BLOB_CONTAINER);
+        filePathOption = optionHolder.validateAndGetOption(Constant.FILE_PATH);
+        appendIfFileExists =
+                Boolean.parseBoolean(optionHolder.validateAndGetStaticValue(Constant.APPEND_IF_FILE_EXISTS, "true"));
+        boolean recreateBlobContainer = Boolean.parseBoolean(
+                optionHolder.validateAndGetStaticValue(Constant.RECREATE_BLOB_CONTAINER, "false"));
+        boolean addToExistingBlobContainer = Boolean.parseBoolean(
+                optionHolder.validateAndGetStaticValue(Constant.ADD_TO_EXISTING_BLOB_CONTAINER, "true"));
+        StorageSharedKeyCredential credential = new StorageSharedKeyCredential(accountName, accountKey);
+        String endpoint = String.format(Locale.ROOT, Constant.ENDPOINT, accountName);
+        storageClient = new DataLakeServiceClientBuilder().endpoint(endpoint).credential(credential).buildClient();
+        log.debug("Successfully connected to Data Lake service and got the storage client for account: " +
```

Comment:
Shall we check isDebugEnabled() since there are string concatenation

Reply:
Fixed with commit https://github.com/siddhi-io/siddhi-io-azuredatalake/pull/1/commits/3599d3fc5cb9b6a659a6f3fe41241a3b7701a668